Application/EdFi.Admin.DataAccess/Providers/AdminDatabaseConnectionStringProvider.cs

```diff
@@ -4,20 +4,42 @@
 // See the LICENSE and NOTICES files in the project root for more information.
 
 #if NETSTANDARD
+using System;
 using EdFi.Ods.Common.Configuration;
+using EdFi.Ods.Common.Context;
 
 namespace EdFi.Admin.DataAccess.Providers
 {
     public class AdminDatabaseConnectionStringProvider : IAdminDatabaseConnectionStringProvider
     {
+        private const string ConnectionStringName = "EdFi_Admin";
+        protected readonly ApiSettings ApiSettings;
         private readonly IConfigConnectionStringsProvider _configConnectionStringsProvider;
+        private readonly IInstanceIdContextProvider _instanceIdContextProvider;
 
-        public AdminDatabaseConnectionStringProvider(IConfigConnectionStringsProvider configConnectionStringsProvider)
+        public AdminDatabaseConnectionStringProvider(ApiSettings apiSettings, IConfigConnectionStringsProvider configConnectionStringsProvider,IInstanceIdContextProvider instanceIdContextProvider)
         {
+            ApiSettings = apiSettings;
             _configConnectionStringsProvider = configConnectionStringsProvider;
+            _instanceIdContextProvider = instanceIdContextProvider;
         }
 
-        public string GetConnectionString() => _configConnectionStringsProvider.GetConnectionString("EdFi_Admin");
+        public string GetConnectionString()
+        {
+            if (ApiSettings.GetApiMode() == ApiMode.InstanceYearSpecific)
+            {
+                string dbName = ConnectionStringName;
+                string instanceId = _instanceIdContextProvider.GetInstanceId();
+                if (string.IsNullOrEmpty(instanceId))
+                {
+                    throw new InvalidOperationException("Expected instanceId is null or empty.");
+                }
+                string connStr = _configConnectionStringsProvider.GetConnectionString(ConnectionStringName).Replace(dbName, $"{dbName}_{instanceId}");
```

Comment:
This is assuming the Database name always has to be "EdFi_Admin".  This would probably be better to match the pattern of ODS, and use a replacement token approach so the implementation can prefix their databases however they want.